Extension/src/LanguageServer/extension.ts

```diff
@@ -293,9 +294,43 @@ function realActivation(): void {
     updateLanguageConfigurations();
 
     reportMacCrashes();
-
+    
     const settings: CppSettings = new CppSettings(clients.ActiveClient.RootUri);
-    if (settings.updateChannel === 'Insiders') {
+    let suggestInsiders: PersistentState<boolean> = new PersistentState<boolean>("CPP.suggestInsiders", true);
+    
+    if (suggestInsiders.Value && settings.updateChannel === 'Default') {
+        getTargetBuildInfo("Insiders").then(buildInfo => {
+            if (!buildInfo) {
+                return;
+            }
+            const message: string = `The insider version (${buildInfo.name}) is available, switch to the insider channel?`;
+            const yes: string = "Yes";
+            const askLater: string = "Ask Me Later";
+            const dontShowAgain: string = "Don't Show Again";
+            vscode.window.showInformationMessage(message, yes, askLater, dontShowAgain).then(selection => {
+                switch (selection) {
+                    case yes:
+                        // Avoid double calling applyUpdate
+                        shouldCheckAndApplyUpdate = false;
+                        vscode.workspace.getConfiguration("C_Cpp").update("updateChannel", "Insiders", true);
+                        return applyUpdate(buildInfo, 'Insiders');
```

Comment:
If I use `onDidChangeSettings` to call `checkAndApplyUpdate`, `getTargetBuildInfo` will be called twice. I want to avoid this situation.

Reply:
Got it. The comment made me think it was for a different reason.  I'm still trying to think of a different way to do this since I feel the code would flow more naturally if the settings change logic managed the update.  Perhaps we can cache the results of `getTargetBuildInfo` and have the update code clear the stored value when an actual update completes (or fails).

Reply:
Ok, I will cache `buildInfo` and refactor the code.